roadmap/implementers-guide/src/node/validity/assignments.md

```diff
@@ -0,0 +1,101 @@
+# Approval assignments
+
+Approval assignment determines on which candidate parachain blocks each validator performs approval checks.  An approval session consideres only one relay chain block and assigns only those candidates that relay chain block declares available.
+
+Assignment balances several concerns:
+
+- limits adversaries foreknowledge about assignments,
+- ensures enough checkers, and 
+- distributes assignments relatively equitably.  
+
+Assignees determine their own assignments to check specific candidates using two or three assignment criteria.  Assignees never reveal their assignments until relevant, and gossip delays assignments sent early, which limits others' foreknowledge.  Assignees learn their assignment only with the relay chain block. 
```

Comment:
When do they reveal their assignments?

Reply:
I'm afraid this first pass focused primarily upon the assignment module's interface, so I forgot that part presumably, but..

We should clarify that assignment noticed go out only when their delay tranche gets reached.  It's more complex actually since they could be delayed further, either by waiting for no shows, or else by legitimately waiting to reduce their workload.  I've spent today coding that portion, which became obnoxiously subtle.  I'll add some text once I've cleaned up this new code.

Reply:
https://github.com/w3f/research-internal/issues/519

Reply:
3bec0174